### PR TITLE
[sqflite] Update sqflite to 2.3.0

### PR DESCRIPTION
**Pull request by Swanseo0**
* Update sqflite to 2.3.0
* Increase the minimum Flutter version to 3.3.
* Increase the minimum sdk version to 3.0.0.

Related to #609

### CODE REVIEW

**Review by JSUYA — APPROVED** [review]
LGTM

File: packages/sqflite/CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,8 @@
-## NEXT
+## 0.2.0
```
[the remark this answers is not in the document]
Comment (Swanseo0) [reply packages/sqflite/CHANGELOG.md]: I must have been mistaken. I don't think we need to increase the minor version.
Increased the patch version. Thank you.

File: packages/sqflite/CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,8 @@
...
 
+* Update sqflite to 2.3.0
 * Increase the minimum Flutter version to 3.3.
+* Increase the minimum sdk version to 3.0.0.
```
[the remark this answers is not in the document]
Comment (Swanseo0) [reply packages/sqflite/CHANGELOG.md]: Fixed typo. Thank you.

File: packages/sqflite/analysis_options.yaml
[the remark this answers is not in the document]
Comment (Swanseo0) [reply packages/sqflite/analysis_options.yaml:1]: Added analyzer's language rules.

**Review by swift-kim — APPROVED** [review]

**Review by JSUYA — COMMENTED** [review]
Can you explain the changes to packages/sqflite/example/lib/batch_test_page.dart?

Some code differs from frontend plugin. (var -> final)
Is it because of the analyzer's `strict-inference` option?

**Comment by Swanseo0** [comment]
> Can you explain the changes to packages/sqflite/example/lib/batch_test_page.dart?
> Some code differs from frontend plugin. (var -> final)
> Is it because of the analyzer's strict-inference option?

@JSUYA Because of the analyzer's `prefer_final_locals`.
https://github.com/tekartik/sqflite/blob/0cda463f3fb39c73ad947ed37a45f9c783b2a179/sqflite/analysis_options.yaml#L99

**Comment by JSUYA** [comment]
> > Can you explain the changes to packages/sqflite/example/lib/batch_test_page.dart?
> > Some code differs from frontend plugin. (var -> final)
> > Is it because of the analyzer's strict-inference option?
> 
> @JSUYA Because of the analyzer's `prefer_final_locals`. https://github.com/tekartik/sqflite/blob/0cda463f3fb39c73ad947ed37a45f9c783b2a179/sqflite/analysis_options.yaml#L99

Thank you for reply. First of all, we don't have this [`analysis_option.yaml`](https://github.com/tekartik/sqflite/blob/master/sqflite/example/analysis_options.yaml) in our example path.
Please add this file and check again.

**Review by JSUYA — APPROVED** [review]

**Pull request merged**